src/app_service/service/transaction/service.nim

```diff
@@ -362,3 +369,69 @@ QtObject:
     except Exception as e:
       error "Error fetching crypto services", message = e.msg
       return @[]
+
+  proc addToAllTransactionsAndSetNewMinMax(self: Service, myTip: float, numOfTransactionWithTipLessThanMine: var int, 
+    transactions: JsonNode) =
+    if transactions.kind != JArray:
+      return
+    for t in transactions:
+      let gasPriceUnparsed = $fromHex(Stuint[256], t{"gasPrice"}.getStr)
+      let gasPrice = parseFloat(wei2gwei(gasPriceUnparsed))
+      if gasPrice < myTip:
+        numOfTransactionWithTipLessThanMine.inc
+
+  proc getEstimatedTime*(self: Service, priorityFeePerGas: string, maxFeePerGas: string): EstimatedTime = 
+    let priorityFeePerGasF = priorityFeePerGas.parseFloat
+    let maxFeePerGasF = maxFeePerGas.parseFloat
+    var transactionsProcessed = 0
+    var numOfTransactionWithTipLessThanMine = 0
+    var latestBlockNumber: Option[Uint256]
+    var expectedBaseFeeForNextBlock: float
+    try:
+      let response = eth.getBlockByNumber("latest", true)
```

Comment:
I think just taking in account the latest block might not reflect an accurate result. Maybe during QA we can compare the results of this formula against these websites, to know how close are we to their estimates?
https://www.gasprice.io/
https://ethgasstation.info/
https://etherscan.io/gastracker

Reply:
@richard-ramos it doesn't take only the last block, but last ~100 transactions, that can be last 5,6,7... 10... blocks.

Reply:
You can have a look to the method: eth fee history to see more block details:
e.g: to calculate gas price suggestion we do:
```
err = f.RPCClient.Call(&feeHistory, chainID, "eth_feeHistory", 101, "latest", nil)
    if err != nil {
        return nil, err
    }
```